Offerel Theme

Offerel opened the pull request:
Closes #

Changes proposed in this pull request:

-
-
-

How to test the feature manually:

1.
2.
3.

Pull request checklist:

- [ ] clear commit messages
- [ ] code manually tested
- [ ] unit tests written (optional if too hard)
- [ ] documentation updated

[Additional information can be found in the documentation](https://github.com/FreshRSS/FreshRSS/tree/edge/docs/en/developers/04_Pull_requests.md).

Commit:
Offerel Theme

Alkarex commented on p/themes/Offerel/metadata.json:
```diff
@@ -0,0 +1,7 @@
+  "author": "Offerel",
+  "description": "A Fork of Swage.",
+  "version": 0.9,
+  "files": ["_template.css", "swage.css", "offerel.css"]
```
If possible, just link to the base Swage theme this way (not tested):

```suggestion
  "files": ["_template.css", "../Swage/swage.css", "offerel.css"]
```

Offerel replied:
I have changed your suggestion, which worked and also removed the now un-needed files from my theme. I got a (automated?) mail, that there was Push initiated. Maybe this was something automated, because i committed the changes to my branch.

Commit:
removed assets

Alkarex committed:
Merge branch 'edge' into Offerel-Theme

Alkarex committed:
make fix-all
And manual fix of duplicate selector ".flux"

math-GH commented:
YFI: I will review it after I fixed the Swage theme, that is used here as basis

math-GH commented:
Let's wait for #4493 and will see what effect it will have on this theme.

Please check the mobile view, too

Alkarex committed:
Merge branch 'edge' into Offerel-Theme

Alkarex commented:
Would you have a chance to look at this PR  @math-GH ?

math-GH committed:
Merge branch 'edge' into Offerel-Theme

math-GH commented:
@Offerel :
- please write some introduction words into the first text field of this PR, so that we would understand your idea behind this new theme
- please do not hide the FreshRSS logo
- please check all menus: the hover color (blue) hides the black font color
- subscription menu button: grey/black font color on orange background is not good readable
- navigation buttons: the buttons do not have always a orange colored hover
- check also the login and registration screens

Alkarex committed:
Merge branch 'edge' into Offerel-Theme

Alkarex committed:
Merge branch 'edge' into Offerel-Theme

Offerel commented:
> @Offerel :
> 
> * please write some introduction words into the first text field of this PR, so that we would understand your idea behind this new theme
> * please do not hide the FreshRSS logo
> * please check all menus: the hover color (blue) hides the black font color
> * subscription menu button: grey/black font color on orange background is not good readable
> * navigation buttons: the buttons do not have always a orange colored hover
> * check also the login and registration screens

I will have a look, but im a little bit busy currently. Will do my best to check it this week.

math-GH committed:
Merge branch 'edge' into Offerel-Theme

Offerel committed:
Merge branch 'FreshRSS:edge' into Offerel-Theme

Offerel commented:
I have tried to fix the issues in the list. Except for the two points with the MouseOver, everything should fit now. I could not find anything about these two points. If you have screenshots, that would certainly be helpful for me or another way to show me where exactly the problems occur.

I have now implemented the dragging of the side panel with javascript, instead of CSS. I think this is more intuitive to use. This was not possible with the first version, because there was no javascript integrated in the theme. Login and account creation pages are also modified.

I am not sure how the changes come to you. I checked out the status on my end, made the changes and re-committed to my branch. Does this update automatically? I am making a new pull request in parallel, if this is not what you need, please close this and tell me how to commit the changes.

Offerel committed:
Merge branch 'edge' into Offerel-Theme

Alkarex commented:
Yes, you can just simply pull / push new commits to this branch (no need to squash, rebase, etc.)

math-GH commented:
I personally like the Github Desktop app. It has a good UI to do the most important things.

Offerel commented:
Mostly I'm using VSCodium (Visual Studio Code) under Linux. Here I can code and use the integrated GIT integration. I have moved my private repositories from GitHub to CodeBerg some months ago and on Work I have to use GitLab. But it should be no problem to open the cloned directory also with GitHub Desktop. I will give it a try tomorrow.

Alkarex commented:
For VSCode, see https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens

Offerel committed:
Merge branch 'edge' into Offerel-Theme

Offerel commented:
I have added modifications for mobile phones

Alkarex commented:
The bottom / footer of articles seems a bit buggy, in particular the sharing part:

![image](https://user-images.githubusercontent.com/1008324/188963935-353b2736-10a2-4876-a818-8c5ce8a5c162.png)

Alkarex commented:
When we scroll a bit down, the hover style over article titles fail:

![image](https://user-images.githubusercontent.com/1008324/188964395-070a095a-92eb-444d-ba19-715bcfb29e16.png)

Offerel commented:
OK, thanks for the screens. I'll take a look tomorrow and send an update.

Offerel commented:
I was able to fix the share menu. Unfortunately, I cannot reproduce the effect with the MouseOver. I have tried this with Edge (Windows), Chromium (Linux/Windows) and Firefox (Linux/Windows) but without success. Another thing that bothers me, is the scrolling behaviour. When I scroll down in the article list, the SidePanel moves as well. At least the logo disappears and the folders move up a bit. I suspect this is a part of the JavaScript somewhere and not theme dependent. I would like to disable this for my theme, as this animation makes me a bit nervous :-) Where in the source code can I find this?

Offerel commented:
For some reason today after sync, i got some strange mouseover effects and colors at some parts. It seems that the Swage theme was updated and my theme didn't reflect that. I have tried to fix that, so that the blue colors do not pop out. Maybe this is what @math-GH was talking about.

In one of the later releases, I can try to make my theme independent from Swage. This should prevent me from such things. But I'm a lazy person.

BTW: I didn't see my latest commits here in this pull request. But since @Alkarex means, that this done automatically, i hope this is the case.

math-GH committed:
Merge branch 'edge' into Offerel-Theme

math-GH commented:
now this theme is completely broken :/

Swage supports now CSS vars. It would be possible now to customize it very easy:

These colors are used by Swage. It would be enough to have just `:root` and customize the colors.

```
:root {
	--color-text-light: #fcfcfc;
	--color-text-light-darker: #969696;
	--color-text-dark: #181621;
	--color-text-nav: #00488b;
	--color-text-aside: #22303d;
	--color-text-alert: #fa8052;
	--color-text-good: #5eaabf;
	--color-text-bad: #b0425b;
	--color-text-bad-lighter: #c46178;
	--color-text-shadow-light: #fcfcfc;
	--color-box-shadow-light: #e3e3e3;
	--color-border-light: #fcfcfc;
	--color-border-light-darker: #e3e3e3;
	--color-border-grey: #e3e3e3;
	--color-border-nav: #00488b;
	--color-border-bad: #b0425b;
	--color-background-light: #fcfcfc;
	--color-background-light-darker: #e3e3e3;
	--color-background-dark: #181621;
	--color-background-nav: #0062be;
	--color-background-nav-darker: #00488b;
	--color-background-aside: #22303d;
	--color-background-alert: #fa8052;
	--color-background-alert-darker: #f95c20;
	--color-background-good: #5eaabf;
	--color-background-bad: #c46178;
	--color-background-stared: #fff6da;
	--color-background-unread: #fff3ed;
	--color-background-hover: #fff;
	--frss-scrollbar-handle: rgba(0, 0, 0, 0.1);
	--frss-scrollbar-handle-hover: rgba(0, 0, 0, 0.4);
	--frss-scrollbar-track: rgba(0, 0, 0, 0.05);
	--frss-scrollbar-track-hover: rgba(0, 0, 0, 0.1);
}
```

Offerel commented:
I ill have a look into this, when im back in december.

math-GH committed:
Merge branch 'edge' into Offerel-Theme

math-GH commented:
Any news here @Offerel ?

Offerel commented:
To be honest, I gave up at that point. The last changes to the base theme have changed the layout so that I would have to completely rewrite my theme. I honestly don't have the desire to do that, as I wouldn't know how to implement it quickly either. I don't even understand the changes that were made and why they were made. I have now first brought in a few of the adjustments via the CustomCSS plugin, so that I have the most important adjustments also now. So these also do not disappear, with an update. This is all the work I wanted to do with it at least for now.

math-GH commented:
Creating and maintaining a theme is not so easy. That is why the CSS vars should help to tweak themes very easy.

Please feel free to close the PR if there is no current work anymore
